### PR TITLE
APPINT-31368 Support for non-clear-text passwords in configuration.

### PR DESCRIPTION
**Pull request by andreasmattes**
The present securtity fix adds enhancements to OSGi and keystore configuration data retrieval to allow for obfuscated passwords or passwords passed through OS environment.

### CODE REVIEW

File: auxiliary-storage/features/src/main/resources/features.xml [hunk 1]
```diff
@@ -37,73 +37,75 @@
...
-    <feature version="${original.version}">tesb-aux-client-rest</feature>
-    <feature version="${original.version}">tesb-aux-service-rest</feature>
-    <feature version="${original.version}">tesb-aux-persistence</feature>
-    <feature version="${original.version}">tesb-aux-server</feature>
-  </feature>
+    <!-- Common -->
+
+    <feature name='tesb-aux-common' version='${original.version}'>
+        <!--External dependency -->
+        <feature version='${cxf.version}'>cxf-rs-security-xml</feature>
```
Comment (jbonofre) [inline auxiliary-storage/features/src/main/resources/features.xml:44]: Just a reminder, version `1.0` actually means `[1.0,)`. If you really want a target version, it's better to use `[1.0,1.0]`.

**Review by jbonofre — APPROVED** [review]
Overall OK, a possible alternative (probably simpler) is to use a ConfigAdmin plugin/service. It's what I did for interpolation in Karaf 4.3.x. Maybe we can use this approach in our runtime when we will upgrade to Karaf 4.3.x.

File: auxiliary-storage/features/src/main/resources/features.xml [hunk 2]
```diff
@@ -20,15 +20,15 @@
   -->
 <features name="tesb-aux-${original.version}" xmlns="http://karaf.apache.org/xmlns/features/v1.0.0">
 
-  <repository>mvn:org.apache.cxf.karaf/apache-cxf/${cxf.version}/xml/features</repository>
-  <repository>mvn:org.apache.camel.karaf/apache-camel/${camel.version}/xml/features</repository>
+    <repository>mvn:org.apache.cxf.karaf/apache-cxf/${cxf.version}/xml/features</repository>
```
[the remark this answers is not in the document]
Comment (andreasmattes) [reply auxiliary-storage/features/src/main/resources/features.xml:23]: Checked to be really just legacy and removed.

File: auxiliary-storage/features/src/main/resources/features.xml (hunk 1 above)
Comment (andreasmattes) [reply auxiliary-storage/features/src/main/resources/features.xml:44]: Replaced by proper version range.

**Review by jgreffe — APPROVED** [review]

**Pull request merged**